Content.Server/BodySystem/Mechanisms/Mechanism.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections.Generic;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+using YamlDotNet.RepresentationModel;
+
+
+
+
+namespace Robust.Shared.BodySystem {
+    public enum MechanismType { passive, toggle, active }
+
+    [Prototype("mechanism")]
+	public class Mechanism {
+		private string _name;
+		private string _description;
+		private string _examineMessage;
+		private int _durability;
+		private int _currentDurability;
+		private int _destroyThreshold;
+		private int _resistance;
+		private int _size;
+		private BodyPartCompatability _compatability;
+		private MechanismType _type;
+		private List<IMechanismEffect> _onTick;
```

Comment:
I feel like a ton of lists of all these possible interactions is kind of a mess.

It might be better to just make mechanisms full blown entities so you can apply the regular component system to them. Or something like that.

Reply:
These are still just placeholder for now, we'll have to figure out a spell system to optimally do this. I'll comment this out for now